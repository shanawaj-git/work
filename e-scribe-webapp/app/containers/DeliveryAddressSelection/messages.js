import { defineMessages } from 'react-intl';

export const scope = 'delivery.address.selection';

export default defineMessages({
  deliveryTitle: {
    id: `${scope}.delivery.title`,
    defaultMessage: 'Where would you like your\nmeds delivered to?',
  },
  deliveryDescription: {
    id: `${scope}.delivery.description`,
    defaultMessage:
      'You donâ€™t have any saved addresses.\nAdd an address to get started.',
  },
  addNewAddress: {
    id: `${scope}.addnew`,
    defaultMessage: 'Add new address',
  },
});
