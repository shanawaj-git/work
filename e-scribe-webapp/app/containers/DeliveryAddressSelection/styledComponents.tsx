import styled from 'styled-components';
import { colorSelector } from 'themes';
import { Typography } from '@albathanext/design-system';

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  flex: 0 0 auto;
  flex-grow: 1;
  flex-shrink: 0;
  flex-basis: auto;
  min-height: 400px;
  flex-direction: column;
  padding: 0px 18px;
  max-width: 100%;
`;

export const Wrapper = styled.div`
  display: flex;
  padding-top: 50px;
  padding-bottom: 48px;
  justify-content: flex-start;
  align-items: fles-start;
  flex-direction: column;
  width: 100%;
`;

export const TabContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;

  .btnAddNew {
    margin-top: 16px;
    width: 221px;
    height: 46px;
    text-transform: none !important;
    font-weight: 600 !important;
    font-size: 17px !important;
    line-height: 22px !important;
    letter-spacing: -0.408px !important;
    font-style: normal !important;
  }
`;

export const DeliveryTitle = styled(Typography)`
  margin-top: 200px !important;
  font-weight: 700 !important;
  letter-spacing: 0.35px !important;
  text-align: center;
  white-space: pre-line;
  color: ${colorSelector('font-primary')};
`;

export const DeliveryDescription = styled(Typography)`
  margin-top: 16px !important;
  font-weight: 400 !important;
  font-size: 15px !important;
  line-height: 20px !important;
  text-align: center;
  letter-spacing: -0.24px !important;
  white-space: pre-line;
  color: ${colorSelector('font-primary')};
`;
