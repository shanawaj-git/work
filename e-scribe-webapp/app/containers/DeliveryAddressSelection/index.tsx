import React from 'react';
import ScrollLock from 'react-scrolllock';
import PropTypes, { InferProps } from 'prop-types';
import { intlShape, injectIntl } from 'react-intl';
import { TypographyType, Button } from '@albathanext/design-system';

import PharmacySelectionTabs, {
  PharmacySelectionTab,
} from 'components/organism/PharmacySelectionTabs';
import {
  Container,
  Wrapper,
  TabContainer,
  DeliveryTitle,
  DeliveryDescription,
} from './styledComponents';
import messages from './messages';

function DeliveryAddressSelection({ intl }: DeliveryAddressSelectionTypes) {
  return (
    <ScrollLock>
      <Container>
        <Wrapper>
          <PharmacySelectionTabs activeTab={PharmacySelectionTab.Delivery} />
          <TabContainer>
            <DeliveryTitle typographyType={TypographyType.TITLE_2}>
              {intl.formatMessage(messages.deliveryTitle)}
            </DeliveryTitle>
            <DeliveryDescription typographyType={TypographyType.FOOT_NOTE}>
              {intl.formatMessage(messages.deliveryDescription)}
            </DeliveryDescription>
            <Button className="btnAddNew" size="medium" variant="contained">
              <>{intl.formatMessage(messages.addNewAddress)}</>
            </Button>
          </TabContainer>
        </Wrapper>
      </Container>
    </ScrollLock>
  );
}

const DeliveryAddressSelectionPropTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
  intl: intlShape,
};

type DeliveryAddressSelectionTypes = InferProps<
  typeof DeliveryAddressSelectionPropTypes
>;
DeliveryAddressSelection.defaultProps = {};
DeliveryAddressSelection.propTypes = DeliveryAddressSelectionPropTypes;

export default injectIntl(DeliveryAddressSelection);
