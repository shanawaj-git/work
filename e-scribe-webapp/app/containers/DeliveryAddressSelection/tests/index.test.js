import React from 'react';
import { render } from 'react-testing-library';
import { BrowserRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';
import DeliveryAddressSelection from '../index';

const MockDeliveryAddressSelection = () => (
  <BrowserRouter>
    <IntlProvider locale="en">
      <DeliveryAddressSelection />
    </IntlProvider>
  </BrowserRouter>
);

describe('<DeliveryAddressSelection />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<MockDeliveryAddressSelection />);
    expect(firstChild).toMatchSnapshot();
  });
});
