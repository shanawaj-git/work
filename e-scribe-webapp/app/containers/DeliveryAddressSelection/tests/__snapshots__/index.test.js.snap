// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeliveryAddressSelection /> should render and match the snapshot 1`] = `
<div
  class="sc-htpNat eIfBKM"
>
  <div
    class="sc-bxivhb dyCLuD"
  >
    <div
      class="sc-bdVaJa doFTJW"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 sc-bwzfXH hoMxZI css-1jfg05l-MuiTypography-root"
      >
        Select Pharmacy
      </h2>
      <div
        class="root MuiTabs-root styledComponents__StyledTabs-sc-16g4axo-1 gllqKx tabs css-1ujnqem-MuiTabs-root"
        id="pharmacy-selection-tabs"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="flexContainer MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth Mui-selected styledComponents__StyledTab-sc-16g4axo-0 iqlSvE css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              data-testid="delivery"
              id="pharmacy-selection-tabs-tab-0"
              role="tab"
              tabindex="0"
              type="button"
            >
              Delivery
              <span
                class="indicator MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root root MuiTab-root MuiTab-textColorPrimary MuiTab-fullWidth styledComponents__StyledTab-sc-16g4axo-0 gcizIM css-1ujykiq-MuiButtonBase-root-MuiTab-root"
              data-testid="clickncollect"
              id="pharmacy-selection-tabs-tab-1"
              role="tab"
              tabindex="-1"
              type="button"
            >
              Click & Collect
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-ifAKCX eRfXQC"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 sc-EHOje hJAwOz css-1ewfuka-MuiTypography-root"
      >
        Where would you like your
meds delivered to?
      </h3>
      <p
        class="MuiTypography-root MuiTypography-body2 sc-bZQynM ivBfrH css-ry76j1-MuiTypography-root"
      >
        You donâ€™t have any saved addresses.
Add an address to get started.
      </p>
      <button
        class="root StyledButton-sc-1apep4b-0 jVvup btnAddNew MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <div
          style="visibility: visible;"
        >
          Add new address
        </div>
      </button>
    </div>
  </div>
</div>
`;
